core/lib.rs

```diff
@@ -579,6 +581,17 @@ impl Connection {
         }
     }
 
+    pub fn close_async(&self) -> impl Future<Output = Result<(), LimboError>> + '_ {
+        future::poll_fn(move |cx| match self.pager.checkpoint() {
+            Ok(CheckpointStatus::Done(_)) => Poll::Ready(Ok(())),
+            Ok(CheckpointStatus::IO) => {
+                cx.waker().wake_by_ref();
```

Comment:
1. isnt this just busy polling - task will be rescheduled immediately?
2. i don't think this will make progress on IO at all if .run_once() is never called